src/persist-client/src/iter.rs

```diff
@@ -85,6 +85,14 @@ impl<T: Codec64 + Timestamp + Lattice> FetchData<T> {
             FetchData::Unleased { part, .. } => part.key.split().0 >= min_version,
             FetchData::Leased { part, .. } => match &part.part {
                 BatchPart::Hollow(x) => x.key.split().0 >= min_version,
+                // Inline parts are only written directly by the user and so may
+                // be unconsolidated.
+                //
+                // WIP: This makes me realize that having the version of the
```

Comment:
Oh @bkirwi thoughts here?

Reply:
It's a good question! Quick thoughts:
- It's not useful for the specific thing that having the writer version has been useful for, or the planned future use of garbage collection.
- The fact that these parts are meant to be short-lived makes it less important. (If we regret the encoding, it's pretty feasible to eg. rewrite all of them instead of keeping compatability.)
- Seems pretty easy to add later.

So I guess while I think it's a good idea and we may need it sooner or later, I'd be tempted to punt for now. But it's a good callout!